### PR TITLE
this should unfuck the fucked up Joystick pr

### PR DESCRIPTION
**Pull request by time-killer-games**
my last pr broke all joystick shit. this goes against my religion. better get fixed right quick.

### CODE REVIEW

**Comment by time-killer-games** [comment]
@RobertBColton please review...

File: ENIGMAsystem/SHELL/Platforms/SDL/Joystick.cpp [hunk 1]
```diff
@@ -79,20 +79,28 @@ namespace enigma {
...
+  if (joystick2 != NULL)
+    SDL_JoystickClose(joystick2);
+  
   SDL_QuitSubSystem(SDL_INIT_JOYSTICK);
 }
   
 void joystick_update() {  
   int joystick_count = SDL_NumJoysticks();
-  if (joystick_count == 0) {
+  if (joystick_count <= 0) {
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Platforms/SDL/Joystick.cpp]: This logic is basically identical to `joystick_uninit`. Please move it there and call it here.

**Comment by gfundies** [comment]
are arrays and for loops also against your religion?

**Comment by time-killer-games** [comment]
@gfundies please educate me on how doing a for loop and an array in place of one small instance in an entire game engine where two variables are being initiallized, please do tell me, how doing that once, inintializinh two variables, will slow down the entire game engine's games 8x slower than GameMaker like what you claimed about 1 misused double. Please do tell me. If the consequence isn't big, it's not worth the fuss. I will fix it if you tell me what's wrong with it, just like I did last time, without any hesitemation. I just didn't like your condensending words, that was my only opposition.

**Pull request merged**